src/lib/libx6adc.cpp

```diff
@@ -6,20 +6,30 @@
  */
 
 #include "headings.h"
-#include "X6_1000.h"
 #include "libx6adc.h"
+#include "X6_1000.h"
 
 // globals
 map<unsigned, std::unique_ptr<X6_1000>> X6s_;
 unsigned numDevices_ = 0;
 
+const char* get_error_msg(X6_STATUS err) {
+	if (errorsMsgs.count(err)) {
+		return errorsMsgs[err].c_str();
+	}
+	else {
+		return "No error message for this status number.";
+	}
+}
+
 // initialize the library --contructor hook in header
 void init() {
 	//Open the logging file
 	FILE* pFile = fopen("libx6.log", "a");
 	Output2FILE::Stream() = pFile;
 
-	numDevices_ = get_num_devices();
+	static Innovative::X6_1000M x6;
```

Comment:
Exactly, that's probably all the way from some II example code.

On Fri, Mar 6, 2015 at 9:56 AM, Blake Johnson <notifications@github.com>
wrote:

> In src/lib/libx6adc.cpp
> <https://github.com/BBN-Q/libx6/pull/6#discussion_r25949142>:
>
> >  // initialize the library --contructor hook in header
> >  void init() {
> >  	//Open the logging file
> >  	FILE* pFile = fopen("libx6.log", "a");
> >  	Output2FILE::Stream() = pFile;
> >
> > -	numDevices_ = get_num_devices();
> > +	static Innovative::X6_1000M x6;
>
> I guess you didn't change that but just copied from get_num_devices().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BBN-Q/libx6/pull/6/files#r25949142>.
>